### DISCUSSION TITLE
单库多表的场景如何支持？

### DISCUSSION THREAD

**Discussion by wangxiangyu** [apache/flink-cdc#930] [Q&A]
单库多表： 数据库中多张表，每个表的DDL不同
这个需求应该还是比较通用的，如果对于单个数据库里面的每张表，都起一个sql作业同步一次binlog，显然成本太高，
这种场景，如何写sql语句支持呢？后续计划如何支持呢？

**Comment by yangrong688** [comment]
flink sql支持正则表达式
stream api 可以支持多表，建议使用 stream api 去实现多表同步的逻辑
具体的可以看下文档
![image](https://user-images.githubusercontent.com/70557521/160240903-45265120-3fa1-4b26-914b-d17bbdf9f6a1.png)

https://ververica.github.io/flink-cdc-connectors/master/content/about.html

**Reply by wangxiangyu** [reply]
嗯，data stream api是可以解决这个问题的。 但是我们当前的集成任务都是flink sql支持的。我们的解决方法补充到了这里

https://github.com/ververica/flink-cdc-connectors/discussions/805#discussioncomment-2720567

**Reply by shadon178** [reply]
flink sql应该做不了吧

**Reply by J-Orange** [reply]
> stream

请问哪里有文档或者demo

**Reply by yangrong688** [reply]
https://ververica.github.io/flink-cdc-connectors/master/content/about.html#usage-for-datastream-api

**Reply by J-Orange** [reply]
> https://ververica.github.io/flink-cdc-connectors/master/content/about.html#usage-for-datastream-api

已经浏览了很多次了这里只有读取数据库变更的api 对于数据获取、处理以及数据如何同步到第三端的api好像并没有 官网对于steamapi的文档太稀缺了 按照官网能落地的还是flinksql的方式

**Comment by GourdErwa** [comment]
> 单库多表： 数据库中多张表，每个表的DDL不同 这个需求应该还是比较通用的，如果对于单个数据库里面的每张表，都起一个sql作业同步一次binlog，显然成本太高， 这种场景，如何写sql语句支持呢？后续计划如何支持呢？

API 方式可以分流，也可以使用 statementSet.attachAsDataStream(); ，让多个 sql 以 DataStream 方式运行在一个任务里。
[可参考 MysqlCDC2ChangeLog](https://github.com/GourdErwa/flink-advanced/blob/main/flink-stream/src/main/java/io/group/flink/stream/cdc/mysql/MysqlCDC2ChangeLog.java#L54)
